test/testdata/dsl/minitest.rb

```diff
@@ -21,11 +21,21 @@ def inside_method
         end
     end
 
+    def instance_helper; end
+
     before do
-        @foo = T.let(3, Integer)
+        @foo = T.let(3, T.nilable(Integer))
```

Comment:
(I made this `nilable` so it can be reset in `after`. I assume that if it understood `after` to be in the wrong scope, it would say that `@foo` was not initialized. If I understand correctly, the DSL pass is injecting a synthetic `initialize`.)